ckanext/xloader/plugin.py

```diff
@@ -183,3 +199,44 @@ def get_helpers(self):
             "xloader_status_description": xloader_helpers.xloader_status_description,
             "is_resource_supported_by_xloader": xloader_helpers.is_resource_supported_by_xloader,
         }
+
+
+def _should_remove_unsupported_resource_from_datastore(res_dict):
+    if not toolkit.asbool(toolkit.config.get('ckanext.xloader.clean_datastore_tables', False)):
+        return False
+    has_url_type = True
+    try:
+        has_url_type = toolkit.asbool(res_dict.get('url_type'))
+    except ValueError:
+        pass
+    return (not XLoaderFormats.is_it_an_xloader_format(res_dict.get('format', u''))
+            and (res_dict.get('url_type') == 'upload'
+                or not has_url_type)
+            and (toolkit.asbool(res_dict.get('datastore_active', False))
+                or toolkit.asbool(res_dict.get('extras', {}).get('datastore_active', False))))
+
+
+def _remove_unsupported_resource_from_datastore(resource_id):
+    """
+    Callback to remove unsupported datastore tables.
+    Controlled by config value: ckanext.xloader.clean_datastore_tables.
+    Double check the resource format. Only supported Xloader formats should have datastore tables.
+    If the resource format is not supported, we should delete the datastore tables.
+    """
+    context = {"ignore_auth": True}
+    try:
+        res = toolkit.get_action('resource_show')(context, {"id": resource_id})
+    except toolkit.ObjectNotFound:
+        log.error('Resource %s does not exist.', res['id'])
```

Comment:
`res` has not been populated, this should use `resource_id`

Reply:
Done